src/main/scala/pirate/Read.scala

```diff
@@ -196,6 +196,7 @@ case object ReadErrorEmpty extends ReadError
 case object ReadErrorNotEnoughInput extends ReadError
 case object ReadErrorTooMuchInput extends ReadError
 case class ShowHelpText(sub: Option[String]) extends ReadError
+case class ShowVersion(version: String) extends ReadError
```

Comment:
These are a bit ugly and aren't really required - but I don't know the current state of the code well enough to understand if it can be avoided at the moment (given the ShowHelpText) I am guessing not - so is probably the right thing for now.